nixos/modules/services/search/quickwit.nix

```diff
@@ -0,0 +1,75 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.quickwit;
+  settingsFormat = pkgs.formats.yaml { };
+  configFile = settingsFormat.generate "quickwit.yaml" cfg.settings;
+in
+{
+  options.services.quickwit = {
+    package = mkOption {
+      type = types.package;
+      description = "Quickwit package to use.";
+      default = pkgs.quickwit;
+    };
+
+    enable = mkEnableOption "Quickwit search engine";
+
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "Environment file to pass to the systemd service.";
+    };
+
+    settings = mkOption {
+      description = "Quickwit settings.";
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          listen_address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = "Address to listen on.";
+          };
+
+          data_dir = mkOption {
+            type = types.path;
+            default = "/var/lib/quickwit";
+            description = "Path to the directory where Quickwit will store its data.";
+          };
+
+        };
+      };
+    };
+  };
+  config = mkIf cfg.enable {
+    users.users.quickwit = {
```

Comment:
Needs to go in `nixos/modules/misc/ids.nix` to create a stable UID and GID.

Reply:
No it is deprecated to do this way (from a previous package I've done I was told that).

Reply:
Can you link me to it, so I can learn about it?

Reply:
I suspect it's https://github.com/NixOS/nixpkgs/pull/216230#discussion_r1105534322. I think that is probably wrong, I will comment there.